gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -470,20 +469,32 @@
   public static final String DEFAULT_COMPACTION_COMPACTOR_CLASS = "gobblin.compaction.mapreduce.MRCompactor";
   public static final String COMPACTION_FILE_SYSTEM_URI = COMPACTION_PREFIX + "file.system.uri";
   public static final String COMPACTION_MR_JOB_TIMEOUT_MINUTES = COMPACTION_PREFIX + "mr.job.timeout.minutes";
-  public static final int DEFAULT_COMPACTION_MR_JOB_TIMEOUT_MINUTES = Integer.MAX_VALUE;
+  public static final long DEFAULT_COMPACTION_MR_JOB_TIMEOUT_MINUTES = Long.MAX_VALUE;
+  public static final String COMPACTION_DATA_VERIFICATION_TIMEOUT_MINUTES =
+      COMPACTION_PREFIX + "data.verification.timeout.minutes";
+  public static final long DEFAULT_COMPACTION_DATA_VERIFICATION_TIMEOUT_MINUTES = Long.MAX_VALUE;
   public static final String COMPACTION_RECOMPACT_FOR_LATE_DATA = COMPACTION_PREFIX + "recompact.for.late.data";
   public static final boolean DEFAULT_COMPACTION_RECOMPACT_FOR_LATE_DATA = false;
   public static final String COMPACTION_DEDUPLICATE = COMPACTION_PREFIX + "deduplicate";
   public static final boolean DEFAULT_COMPACTION_DEDUPLICATE = true;
-  public static final String COMPACTION_JOB_LATE_DATA_MOVEMENT_TASK = COMPACTION_PREFIX + "job.late.data.movement.task";
-  public static final String COMPACTION_JOB_LATE_DATA_FILES = COMPACTION_PREFIX + "job.late.data.files";
-  public static final String COMPACTION_COMPLETE_FILE_NAME = "_COMPACTION_COMPLETE";
-  public static final String COMPACTION_LATE_FILES_DIRECTORY = "late";
-  public static final String COMPACTION_ENABLE_SUCCESS_FILE = "mapreduce.fileoutputcommitter.marksuccessfuljobs";
   public static final String COMPACTION_OVERWRITE_OUTPUT_DIR = COMPACTION_PREFIX + "overwrite.output.dir";
   public static final boolean DEFAULT_COMPACTION_OVERWRITE_OUTPUT_DIR = false;
-  public static final String COMPACTION_JARS = COMPACTION_PREFIX + "jars";
-  public static final String COMPACTION_TRACKING_EVENTS_NAMESPACE = COMPACTION_PREFIX + "tracking.events";
+  public static final String COMPACTION_DATA_COMPLETENESS_VERIFICATION_THREAD_POOL_SIZE =
```

Comment:
That's the part I don't get. What does it buy us? We have so many of these options (363 in my version)  that I doubt anyone will say "Let me see what knobs and levers are out there". More likely, the question is "I need to figure out how to configure component X". Given that the property naming is not very consistent, I think the starting point is always going to be the implementation. I think ConfigurationKeys should be only for keys that are shared across components. This is most likely state properties and not real configuration properties.

I also suggest at renaming the configuration properties so that they share a more specific prefix than compaction, e.g. compaction.completeness_verification or similar. Then, "compaction.should.verify.data.completeness" can become "compaction.completeness_verification.enabled", "compaction.data.completeness.verification.class" can become "compaction.completeness_verification.verificatier_class" and so on. I find the current "English phrase with dots" naming scheme confusing as the same concepts can be expressed in different ways.

This also reminds me of another issue -- please add javadocs for all configuration properties what they mean. Looking just at the configs I don't think I can guess what "compaction.num.datasets.verified.together" does.